packages/diracx-web-components/src/components/JobMonitor/jobDataService.ts

```diff
@@ -7,18 +7,43 @@ import utc from "dayjs/plugin/utc";
 dayjs.extend(utc);
 import { fetcher } from "../../hooks/utils";
 import { Filter, SearchBody, Job, JobHistory } from "../../types";
+import type { JobSummary } from "../../types";
 
 function processSearchBody(searchBody: SearchBody) {
   searchBody.search = searchBody.search?.map((filter: Filter) => {
     if (filter.operator == "last") {
-      return {
-        parameter: filter.parameter,
-        operator: "gt",
-        value: dayjs()
-          .subtract(1, filter.value as "hour" | "day" | "month" | "year")
-          .toISOString(),
-        values: filter.values,
-      };
+      const valueStr = filter.value as string;
+      const match = valueStr.match(/^(\d+)\s*(minute|hour|day|month|year)s?$/i);
+
+      if (match) {
+        const amount = parseInt(match[1], 10);
+        const unit = match[2].toLowerCase() as
+          | "minute"
+          | "hour"
+          | "day"
+          | "month"
+          | "year";
```

Comment:
These values are repeated multiple times here, can't we add them in a variable and reuse the variable?